lib/rubocop/cop/lint/missing_super.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for the presence of constructors and lifecycle callbacks
+      # without calls to `super`.
+      #
+      # @example
+      #   # bad
+      #   class Employee < Person
+      #     def initialize(name, salary)
+      #       @salary = salary
+      #     end
+      #   end
+      #
+      #   # good
+      #   class Employee < Person
+      #     def initialize(name, salary)
+      #       super(name)
+      #       @salary = salary
+      #     end
+      #   end
+      #
+      #   # bad
+      #   class Parent
+      #     def self.inherited(base)
+      #       do_something
+      #     end
+      #   end
+      #
+      #   # good
+      #   class Parent
+      #     def self.inherited(base)
+      #       super
+      #       do_something
+      #     end
+      #   end
+      #
+      class MissingSuper < Base
+        CONSTRUCTOR_MSG = 'Call `super` to initialize state of the parent class.'
+        CALLBACK_MSG    = 'Call `super` to invoke callback defined in the parent class.'
+
+        STATELESS_CLASSES = %w[BasicObject Object].freeze
+        CALLBACK_NAMES = %i[inherited included extended prepended].to_set.freeze
```

Comment:
Probably we can have a better name here, but I don't have a solid proposal - e.g. `ClassLifecycleCallbacks` or something along those lines.